Make varlink API more idiomatic

larskarlitski opened the pull request:
See individual commits. This is a first round of fixes I've found when I started working with this API for [cockpit-podman](https://github.com/cockpit-project/cockpit-podman). I have some more, but these were the ones I could easily implement.

I hope that you don't consider the varlink API to be stable, yet? As far as I know, cockpit-podman is the only user so far and I'm fine updating that on the next release.

larskarlitski commented:
Hm, I don't understand the latest error on the "gating" check. It says "9 commits to fix", but not what the fixes should be...

mheon commented on cmd/podman/varlink/io.podman.varlink:
```diff
@@ -393,34 +372,29 @@ type Runlabel(
-# structure.  A [ContainerNotFound](#ContainerNotFound) error will be returned if the container cannot be found.
-# See also [ListContainers](ListContainers) and [InspectContainer](#InspectContainer).
-method GetContainer(name: string) -> (container: ListContainerData)
+# Returns information about a single container.  If a container with the given
```
I think we'd prefer that all descriptions start with the name of the API endpoint, for consistency (so `GetContainer returns...` instead)

larskarlitski replied:
Ooh I hadn't noticed that, but I like it. Changed the ones I touched to be consistent again.

mheon commented on pkg/varlinkapi/system.go:
```diff
@@ -15,22 +16,14 @@ func (i *LibpodAPI) GetVersion(call iopodman.VarlinkCall) error {
-
-// Ping returns a simple string "OK" response for clients to make sure
-// the service is working.
-func (i *LibpodAPI) Ping(call iopodman.VarlinkCall) error {
```
@baude Any reason to keep this? I think it might be useful for a liveliness check

baude replied:
yes, do not remove it; we often use it to test that the connection is working.

larskarlitski replied:
Hm, I don't feel super strong about it, but it seems weird to me to have a debug call like this in the API. Could you instead use `GetVersion()` for testing that the connection is working?

mheon commented:
Gitvalidation is complaining that you need a signoff on your commits

@baude Will the time format changes here mean we need changes in `adapter` to read times back in as RFC3339?

mheon commented:
/ok-to-test
bot, add author to whitelist

baude commented:
@larskarlitski cockpit is not the only user of the API ... the remote client for darwin I am writing is the a major one.  Unfortunately we don't quite have integration tests running for those yet as we are still working on a couple of container related ones.  Until that is resolved, the tests cannot be run.  So you will have to make sure you don't break that either.

baude commented:
please dont remove endpoints that are not implemented.  it is the only list we have for what needs to be done still.

baude commented on cmd/podman/varlink/io.podman.varlink:
```diff
@@ -8,7 +8,7 @@ type Version (
   go_version: string,
   git_commit: string,
-  built: int,
+  built: string, # as RFC3339
```
can you verify this still works with the remote client?

larskarlitski replied:
Sure. Where can I find this client?

baude commented on cmd/podman/varlink/io.podman.varlink:
```diff
@@ -386,9 +376,15 @@ type Runlabel(
 
-# GetVersion returns a Version structure describing the libpod setup on their
-# system.
-method GetVersion() -> (version: Version)
```
please add API documentation like many of the other functions. thanks.

larskarlitski replied:
Added it back.

baude commented:
@larskarlitski great start ... i like the changes for images and containers naming. if you can address my comments, that would be great.

rhatdan commented:
@larskarlitski Have you had a chance to address @baude Issues?

larskarlitski commented:
> Gitvalidation is complaining that you need a signoff on your commits

Thanks. Added those and addressed @baude's comments. Thanks for the reviews!

> please dont remove endpoints that are not implemented. it is the only list we have for what needs to be done still.

We can't commit to a stable API if some of the functions return a `NotImplemented` type. Better to leave out the things that aren't there yet. to track missing functions, would you be ok with me adding a list in a `TODO` file or similar?

baude commented:
> > Gitvalidation is complaining that you need a signoff on your commits
> 
> Thanks. Added those and addressed @baude's comments. Thanks for the reviews!
> 
> > please dont remove endpoints that are not implemented. it is the only list we have for what needs to be done still.
> 
> We can't commit to a stable API if some of the functions return a `NotImplemented` type. Better to leave out the things that aren't there yet. to track missing functions, would you be ok with me adding a list in a `TODO` file or similar?

there are several i dont mind pruning.  Ill remove those in a different PR.  Can we compromise and instead of removing, comment them out?

larskarlitski commented:
> there are several i dont mind pruning. Ill remove those in a different PR. Can we compromise and instead of removing, comment them out?

Good idea! Done.

Are these test failures related to my changes? I can't see how they would be...

mheon commented:
Restarted one of the cirrus tests to see if this is just flakes, or we have a real issue

@baude Mind taking another look at this?

baude commented:
kind of baffled by this one; the tests look like their legitimately failing?

larskarlitski commented:
> kind of baffled by this one; the tests look like their legitimately failing?

I don't understand all the tests and it's hard to get them working locally. I've found one place where the remoteClient panics in `splitStringDate`, because it still assumed the old time format. I could remove this function, because we're now just passing along standard time stamps.

rhatdan commented:
@baude @jwhonce Tests are passing PTAL

baude commented:
please rebase and I will merge it @larskarlitski

larskarlitski commented:
Rebased.

larskarlitski commented:
Rebased, again.

baude commented:
@larskarlitski I will watch your PR .. .btw, join us on #podman on freenode!

baude commented:
/approved

baude commented:
/lgtm
